src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -257,6 +254,24 @@ protected CliGitAPIImpl(String gitExe, File workspace,
         this.environment = environment;
 
         launcher = new LocalLauncher(IGitAPI.verbose?listener:TaskListener.NULL);
+
+        GitTool git = resolveGitTool();
+        if (git != null && git.isHideDebugCommands()) {
+            this.verbose = false;
+        }
+    }
+
+    public GitTool resolveGitTool() {
```

Comment:
Would you consider adding the findbugs suppression for this line, as shown in the [evaluation branch](https://github.com/MarkEWaite/git-client-plugin/commit/36a8abe8d29d0dec272395bb34b336a79063ddd0) I was using?

I haven't found a way to avoid that findbugs suppression.  The tests are structured in a way that causes GitTool to be used in contexts where there is no jenkins instance.  I don't want to rework the tests, so it is simplest to suppress the findbugs warning on the unnecessary null check when running in a real Jenkins environment.

Reply:
Hi @MarkEWaite. Sorry, this implementation doesn't work properly. Actually, the Jenkins instance is null on a slave node. I need to move this code to make this work.
I'll let you know.